gmso/external/convert_dataframe.py

```diff
@@ -0,0 +1,552 @@
+"""Module support for converting to/from Pandas DataFrame objects."""
+
+import functools
+import warnings
+from collections.abc import Iterable
+from operator import attrgetter, itemgetter
+
+import numpy as np
+import pandas as pd
+import unyt as u
+from symengine import expand
+
+from gmso import Topology
+from gmso.abc.serialization_utils import unyt_to_dict
+from gmso.core.views import PotentialFilters
+
+pfilter = PotentialFilters.UNIQUE_PARAMETERS
+from gmso.exceptions import GMSOError
+from gmso.utils.io import import_
+
+pd = import_("pandas")
+
+
+def to_dataframeDict(
+    topology: Topology,
+    parameter: str = "all",
+    format: str = "default",
+    columns: list[str] = None,
+    handle_unyts: str = "in_headers",
+) -> pd.DataFrame:
+    """Return a dictionary of pandas dataframe objects for a topology.
+
+    Parameters
+    ----------
+    topology : gmso.Topology, required
+        Topology to use for converting values
+    parameter : str, optional, default='all'
+        A string determining what aspects of the gmso topology will be reported.
+        Options are: 'all', 'sites', 'bonds', 'angles', 'dihedrals', and 'impropers'. Defaults to 'all'.
+    format : str, optional, default='default'
+        The output formatting style for the dataframe.
+        Options are 'default', 'specific_columns', 'publication', `remove_duplicates`. Defaults to 'default'
+        'default' will output default column values of ["name", "atom_type.name", "atom_type.parameters", "charge", "mass"],
+        and any additional attributes in the `columns` argument.
+        'specific_columns' will only output the attributes from the `columns` argument.
+        'publication' will use the default outputs, but remove duplicate values from the dataframes. It adds a column labeled
+        'Atom Indices' to the `sites` dataframe, which enumerates the indices that the atom_type is a part of.
+        `remove_duplicates` will use the labels in passed through the columns argument, and remove duplicates rows in the dataframe.
+        You must also set the `parameter` argument to be one of {"sites", "bonds", "angles", "dihedrals", "impropers"}, not {"all"}
```

Comment:
I'm kind of confused about this line, and I don't see anything in the notes about it. Is this saying that `remove_duplicates` specifically requires that you don't use `all`

Reply:
Yeah it's not the best I'd agree. How is this?
```
`remove_duplicates` will remove duplicate rows from the dataframe. For sites, this column is `atom_types.name`. For connections, it is the `connection_types.connection_members`. For sites, an additional column will be added, labeled `Atom Indices` that includes the site indexes of members that are the given `atom_type.name`. Because these methods are specific to a given Topology element, the `parameter` argument must be one of {"sites", "bonds", "angles", "dihedrals", "impropers"}, not {"all"}
        See Notes for more details on what this looks like.
```
TODO: I still think an notes example would be helpful.